### PR TITLE
Add python 3.9

### PR DESCRIPTION
**Pull request by nvaytet**

### CODE REVIEW

File: .azure-pipelines/templates/build_and_package.yml [hunk 1]
```diff
@@ -14,18 +14,24 @@ jobs:
...
-          vm_image: 'macOS-10.14'
-          py_version: '3.7'
-        MacOS38:
+          py_version: '3.9'
+        # MacOS37:
+        #   vm_image: 'macOS-10.14'
+        #   py_version: '3.7'
+        # MacOS38:
+        #   vm_image: 'macOS-10.14'
+        #   py_version: '3.8'
```
Comment (OwenArnold) [inline .azure-pipelines/templates/build_and_package.yml:31]: This commented out code is just temporary to perform a single Python version compatibility check?

File: conda_build_config.yaml [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
-liblapack:
-  - 3.8.0
+# liblapack:
+#   - 3.8.0
+
+# _openmp_mutex:
+#   - 4.5 1_gnu
+
+# openblas:
+#   - 0.3.5.*
```
Comment (OwenArnold) [inline conda_build_config.yaml:11]: Removing these requirements and pinned version works locally? I see builds pass, so maybe enough.

File: .azure-pipelines/templates/build_and_package.yml (hunk 1 above)
Comment (nvaytet) [reply .azure-pipelines/templates/build_and_package.yml]: Yes was just for testing

File: .azure-pipelines/pull_request.yml [hunk 3]
```diff
@@ -12,5 +12,5 @@ pr:
 extends:
   template: templates/build_and_package.yml
   parameters:
-    git_rev: "46a08124a610869112b890123bb3a4c5e14486e7"
-    mantid_ver: "6.0.20210412.0948" # calculated from tools/make_version.py/
+    git_rev: "v6.2.0"
```
Comment (nvaytet) [inline .azure-pipelines/pull_request.yml:15]: :+1:

File: .azure-pipelines/templates/build_and_package.yml [hunk 4]
```diff
@@ -20,12 +20,18 @@ jobs:
         Linux38:
           vm_image: 'ubuntu-20.04'
           py_version: '3.8'
+        inux39:
```
Comment (nvaytet) [inline .azure-pipelines/templates/build_and_package.yml:23]:
```suggestion
        Linux39:
```

File: conda_build_config.yaml [hunk 5]
```diff
@@ -1,11 +1,8 @@
...
+  - 3.7
 
-liblapack:
-  - 3.8.0
-
-_openmp_mutex:
-  - 4.5 1_gnu
-  
 boost:
   - 1.75.* # see https://github.com/scipp/scipp/issues/1888
```
Comment (nvaytet) [inline conda_build_config.yaml:5]: In scipp we are pinning to `1.76` i think... this seems inconsistent?

Comment (OwenArnold) [reply conda_build_config.yaml:5]: scipp only uses boost as a build dependency. For mantid it's both a build and run dependency. I believe that 1.75 is as far as Mantid is compatible with boost. I'll double check the native mantid conda recipe though.

**Comment by OwenArnold** [comment]
@nvaytet I think this is good to go. Should just be 7,8,9 todo from the checklist https://scipp.github.io/scippneutron/developer/deployment.html

Once we have the new version we should download on a few systems and check it still works nicely with `scippneutron`.

**Comment by nvaytet** [comment]
looks good to me also

**Pull request merged**